### PR TITLE
feat:internvl test

### PR DESCRIPTION
**Pull request by fjaijf**
#### Before submitting

- [x] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [x] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### Description

模型前20个token前向精度对齐，log_diff为0.039
模型前10个token生成一致
模型训练loss为
[2026-05-11 15:17:30,542] [    INFO] - loss: 0.5230419, learning_rate: 1e-06, global_step: 1, current_memory_allocated: 15.747925996780396, current_memory_reserved: 15.987260103225708, max_memory_allocated: 15.747925996780396, max_memory_reserved: 15.987260103225708, interval_runtime: 2.7301, interval_samples_per_second: 1.4652, interval_steps_per_second: 0.3663, ppl: 1.687151999535022, progress_or_epoch: 0.04, cpu_used_memory: 31.13, cpu_available_memory: 157.4
[2026-05-11 15:17:30,542] [    INFO] - [DataLoad global_step: 1] data_load_time: 30.87 ms (accumulated over 4 micro-batches)
[2026-05-11 15:17:32,202] [    INFO] - loss: 0.6030004, learning_rate: 2e-06, global_step: 2, current_memory_allocated: 15.747925996780396, current_memory_reserved: 17.457413911819458, max_memory_allocated: 16.787208557128906, max_memory_reserved: 17.457413911819458, interval_runtime: 1.6597, interval_samples_per_second: 2.4101, interval_steps_per_second: 0.6025, ppl: 1.8275940955694623, progress_or_epoch: 0.08, cpu_used_memory: 31.11, cpu_available_memory: 157.43
[2026-05-11 15:17:32,202] [    INFO] - [DataLoad global_step: 2] data_load_time: 1.42 ms (accumulated over 4 micro-batches)
[2026-05-11 15:17:33,828] [    INFO] - loss: 0.52101994, learning_rate: 3e-06, global_step: 3, current_memory_allocated: 15.747925996780396, current_memory_reserved: 17.457413911819458, max_memory_allocated: 16.787208557128906, max_memory_reserved: 17.457413911819458, interval_runtime: 1.626, interval_samples_per_second: 2.46, interval_steps_per_second: 0.615, ppl: 1.6837440921652866, progress_or_epoch: 0.12, cpu_used_memory: 31.1, cpu_available_memory: 157.44
[2026-05-11 15:17:33,829] [    INFO] - [DataLoad global_step: 3] data_load_time: 1.43 ms (accumulated over 4 micro-batches)
[2026-05-11 15:17:35,454] [    INFO] - loss: 0.45025554, learning_rate: 4e-06, global_step: 4, current_memory_allocated: 15.747925996780396, current_memory_reserved: 17.457413911819458, max_memory_allocated: 16.787208557128906, max_memory_reserved: 17.457413911819458, interval_runtime: 1.6257, interval_samples_per_second: 2.4605, interval_steps_per_second: 0.6151, ppl: 1.568713003196346, progress_or_epoch: 0.16, cpu_used_memory: 31.25, cpu_available_memory: 157.29
[2026-05-11 15:17:35,454] [    INFO] - [DataLoad global_step: 4] data_load_time: 1.53 ms (accumulated over 4 micro-batches)
[2026-05-11 15:17:37,080] [    INFO] - loss: 0.4761152, learning_rate: 5e-06, global_step: 5, current_memory_allocated: 15.747925996780396, current_memory_reserved: 17.457413911819458, max_memory_allocated: 16.787208557128906, max_memory_reserved: 17.457413911819458, interval_runtime: 1.6263, interval_samples_per_second: 2.4596, interval_steps_per_second: 0.6149, ppl: 1.609808455210971, progress_or_epoch: 0.2, cpu_used_memory: 31.37, cpu_available_memory: 157.17
[2026-05-11 15:17:37,080] [    INFO] - [DataLoad global_step: 5] data_load_time: 1.42 ms (accumulated over 4 micro-batches)
[2026-05-11 15:17:38,697] [    INFO] - loss: 0.42323393, learning_rate: 6e-06, global_step: 6, current_memory_allocated: 15.747925996780396, current_memory_reserved: 17.457413911819458, max_memory_allocated: 16.787208557128906, max_memory_reserved: 17.457413911819458, interval_runtime: 1.6173, interval_samples_per_second: 2.4733, interval_steps_per_second: 0.6183, ppl: 1.5268914399060243, progress_or_epoch: 0.24, cpu_used_memory: 31.38, cpu_available_memory: 157.16

### CODE REVIEW

File: paddleformers/transformers/internvl3/modeling_intern_vit.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+        if not return_dict:
+            return (last_hidden_state, pooled_output) + encoder_outputs[1:]
+
+        return BaseModelOutputWithPooling(
+            last_hidden_state=last_hidden_state,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
```
Comment (liuhao2638) [inline paddleformers/transformers/internvl3/modeling_intern_vit.py]: P1 **优先级：P1**

`InternVisionModel` 默认会走 `return_dict=False`，因为 `PretrainedConfig.use_return_dict` 默认是 `False`。这种情况下 `self.encoder(...)` 返回的是 tuple，下一行访问 `encoder_outputs.last_hidden_state` 会直接触发 `AttributeError`，导致单独加载/调用 `InternVisionModel` 或 `AutoModel` 加载 `intern_vit_6b` 时无法前向。请先按 `return_dict` 分支读取 encoder 输出，或固定让 encoder 返回 dict。

```suggestion
        encoder_outputs = self.encoder(
            inputs_embeds=hidden_states,
            output_hidden_states=output_hidden_states,
            return_dict=return_dict,
        )
        if return_dict:
            last_hidden_state = encoder_outputs.last_hidden_state
            pooled_output = last_hidden_state[:, 0, :]
            return BaseModelOutputWithPooling(
                last_hidden_state=last_hidden_state,
                pooler_output=pooled_output,
                hidden_states=encoder_outputs.hidden_states,
                attentions=encoder_outputs.attentions,
            )

        last_hidden_state = encoder_outputs[0]
        pooled_output = last_hidden_state[:, 0, :]
        return (last_hidden_state, pooled_output) + encoder_outputs[1:]
```

File: paddleformers/transformers/internvl3/modeling_internvl_chat.py [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+        elif (
+            do_sample is None
+            and getattr(generation_config, "do_sample", False)
+            and "decode_strategy" not in generate_kwargs
+        ):
+            generate_kwargs["decode_strategy"] = "sampling"
+
+        num_beams = generate_kwargs.get("num_beams", getattr(generation_config, "num_beams", None))
+        if num_beams is not None and int(num_beams) > 1 and "decode_strategy" not in generate_kwargs:
+            generate_kwargs["decode_strategy"] = "beam_search"
```
Comment (liuhao2638) [inline paddleformers/transformers/internvl3/modeling_internvl_chat.py]: P2 **优先级：P2**

这里把 `num_beams > 1` 自动切到 `beam_search`，但下面只把融合后的 `inputs_embeds` 传给 `self.language_model.generate()`。PaddleFormers 的默认 `expand_inputs_for_generation()` 只扩展 `input_ids/attention_mask` 等字段，不会扩展 `inputs_embeds`，因此 beam search 会把 `input_ids` 变成 `batch_size * num_beams`，而 `inputs_embeds` 仍是原 batch，首轮 forward 就会出现 batch 维不一致。请不要在当前实现里暴露 beam search，或为该路径补齐 `inputs_embeds` 的 beam 扩展。

可按下面这种形状先阻止不支持的路径，避免用户调用后在深层报形状错：

```python
        if num_beams is not None and int(num_beams) > 1:
            raise NotImplementedError("InternVLChatModel.generate currently does not support beam search with inputs_embeds.")
```

**Review by liuhao2638 — CHANGES_REQUESTED** [review]
已完成本轮审查，发现一个需要先修复的阻塞问题：`InternVisionModel` 默认前向路径会在 `return_dict=False` 时崩溃。另有一个生成接口的 beam search 兼容性问题也已标在行级评论中。详细建议请看 inline review comments。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Review by risemeup1111 — COMMENTED** [review]
| 序号 | 位置 | 优先级 | 状态 |
| --- | --- | --- | --- |
| 1 | [`modeling_internvl_chat.py`](https://github.com/PaddlePaddle/PaddleFormers/pull/4416#discussion_r3612252996) | P2 | ✅ |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: paddleformers/transformers/internvl3/modeling_internvl_chat.py [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+        return_dict: Optional[bool] = None,
+        **kwargs,
+    ) -> Union[Tuple, CausalLMOutputWithPast]:
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if input_ids is None:
+            raise ValueError("input_ids must be provided for InternVLChatModel.")
+
+        input_embeds = self.language_model.get_input_embeddings()(input_ids).clone()
+        if pixel_values is not None:
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3/modeling_internvl_chat.py]: P2 **优先级：P2**

这里的视觉特征合并没有区分 cache 场景。`forward()` 公开接收 `past_key_values`，但增量解码的后续步通常只传新 token；如果调用方继续传入同一份 `pixel_values`，这里会重新 `extract_feature()` 并要求当前 `input_ids` 里的 `<IMG_CONTEXT>` 数量和整张图的特征数一致，结果会触发 `Image features and image tokens do not match`，也让 cache 分支无法复用首轮已经合并过的视觉特征。建议在已有 cache 时跳过视觉重合并；如果该 wrapper 不打算支持 `forward(..., past_key_values=...)`，也请在入口显式拒绝这个组合。

**处理要求：请针对该评论进行回复（同意并已修改请回复 Done，不同意请说明理由）。**

```suggestion
        if pixel_values is not None and past_key_values is None:
```

**Review by risemeup1111 — COMMENTED** [review]
已复查当前提交。早前提到的 `return_dict=False` 前向问题和不支持 beam search 的处理在当前 head 仍保持修复；这轮补了一条非阻塞的 cache 分支行级建议，请在对应 inline comment 下回复。

CI 仍有部分 GPU/XPU/集成任务未完成，合入前建议等全部检查结束。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: paddleformers/transformers/internvl3/modeling_internvl_chat.py (hunk 3 above)
Comment (fjaijf) [reply paddleformers/transformers/internvl3/modeling_internvl_chat.py]: Done

Comment (risemeup1111) [reply paddleformers/transformers/internvl3/modeling_internvl_chat.py]: 已确认，当前提交已在 cache 分支跳过视觉特征的重复提取与合并，该问题已修复。

**Review by risemeup1111 — APPROVED** [review]
